content/posts/2021-03-08-why-you-should-use-a-ui-library/2021-03-08-why-you-should-use-a-ui-library.mdx

```diff
@@ -0,0 +1,70 @@
+---
+title: Why I should use a UI library?
+date: '2020-05-17T13:37:03.284Z'
+author: andre
+---
+
+In this post, I want to point out what advantages you can gain from an
+open-source UI library and how it can make your life easier.
+
+## Material Design Tab Example
+
+A while ago I decided to rebuild a React Tab component from Material-UI with
+Vanilla Javascript to get an overview of the complexity and the effort. My main
+goal there was to focus on accessibility like navigation with arrow keys tab
+index handling. Here's the result:
+
+<iframe
+  height="265"
+  style="width: 100%;"
+  scrolling="no"
+  title="Material Design Tab - vanilla JS"
+  src="https://codepen.io/andrew_secret/embed/XWNPJVv?height=265&theme-id=light&default-tab=result"
+  frameborder="no"
+  loading="lazy"
+  allowtransparency="true"
+  allowfullscreen="true"
+>
+  See the Pen{' '}
+  <a href="https://codepen.io/andrew_secret/pen/XWNPJVv">
+    Material Design Tab - vanilla JS
+  </a>{' '}
+  by André Rusakow (
+  <a href="https://codepen.io/andrew_secret">@andrew_secret</a>) on{' '}
+  <a href="https://codepen.io">CodePen</a>.
+</iframe>
+
+I came to my conclusion that it takes a lot of effort and knowledge to build a
+rock-solid UI component from scratch. Especially when it comes to:
+
+- accessibility
+- responsive design
+- edges cases
+- browser compatibility
+
+Some details like tab focus with arrow keys haven't been on my mind at that
+time. Due to my lack of knowledge, I wasn't even capable to fulfill all the
+points you see above. If you compare my example with the Material UI component
+you'll see that there is much more logic in place which makes their UI component
+even more rock-solid.
+
+## Don't Reinvent The Wheel
+
+You don't have to write each time the same code when you start a new project. I
+often underestimate the complexity of a seemingly easy UI component. In most
+cases, I decide to use an open-source library. The main advantages for me are:
+
+- don't have to write and maintain complex javascript logic
+- fewer bugs in a project
+- Accessibility is taken into account
+- UI is way more solid and covers edge cases for me
+
```

Comment:
Could be also interesting: The power of the open-source community. Hundreds of people polishing components, finding bugs and fixing them. Is the UI component really your core business or should you rather focus on the problem you really wanna solve.

Reply:
You basically mentioned all the points, maybe it's just cool to point out the strengths of open-source communities.

Reply:
Totally agree